app/trip_details/visit_shop.py

```diff
@@ -0,0 +1,89 @@
+from app.instances.customer import Customer
+from app.instances.shop import Shop
+
+
+def find_distance_to_shop(
+        customer_location: list,
+        shop_location: list
+) -> int | float:
+    return (
+        (customer_location[0] - shop_location[0]) ** 2
+        + (customer_location[1] - shop_location[1]) ** 2
+    ) ** 0.5
+
+
+def calculate_fuel_cost_one_way(
+        fuel_price: int | float,
+        car_fuel_consumption: int | float,
+        distance: int | float
+) -> int | float:
+    return fuel_price * car_fuel_consumption / 100 * distance
+
+
+def calculate_product_cart_cost(
+        products_to_buy: dict,
+        products_costs: dict
+) -> int | float:
+    total_cost = 0
+    for product_name, quantity in products_to_buy.items():
+        if product_name in products_costs:
+            total_cost += products_costs.get(product_name) * quantity
+    return total_cost
```

Comment:
Use sum() function

Reply:
Thank you for highlighting my mistakes! 
Now, code definitely looks better.
Hope I made right changes.